cli/coredump/coredump.go

```diff
@@ -0,0 +1,177 @@
+	}
+	tarReader, err := MakeTarGzReader(archive)
+	if err != nil {
+		return err
+	}
+	for {
+		header, err := tarReader.Next()
+
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			continue
+		case tar.TypeReg:
+			var pos int
+			// Some archives have strange order of objects,
+			// so we check that all folders exist before
+			// creating a file.
+			pos = strings.LastIndex(header.Name, "/")
+			if pos == -1 {
+				pos = 0
+			}
+			if _, err := os.Stat(header.Name[0:pos]); os.IsNotExist(err) {
+				// 0755:
+				//    user:   read/write/execute
+				//    group:  read/execute
+				//    others: read/execute
+				os.MkdirAll(header.Name[0:pos], 0755)
+			}
+			outFile, err := os.Create(header.Name)
+			if err != nil {
+				return err
+			}
+			if _, err := io.Copy(outFile, tarReader); err != nil {
+				return err
+			}
+			outFile.Close()
+
+		default:
+			return fmt.Errorf("uknown type: %b in %s", header.Typeflag, header.Name)
+		}
+
+	}
+	archive.Close()
+	return nil
+}
+
+// Pack packs coredump into a tar.gz archive.
+func Pack(coreName string) error {
+	corePath, err := filepath.Abs(coreName)
+	if _, err := os.Stat(corePath); err != nil {
+		return fmt.Errorf("There was some problem packing archive. "+
+			"Error: '%v'.", err)
+	}
+	content, err := util.ReadEmbedFile(corescripts, "scripts/tarabart.sh")
+	if err != nil {
+		return err
+	}
+	cmd := exec.Command("sh", "-s")
+	StdinPipe, err := cmd.StdinPipe()
+	if err != nil {
+		return err
+	}
+	cmd.Env = append(cmd.Env, "COREFILE_ENV="+corePath)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err = cmd.Start()
+	StdinPipe.Write([]byte(content))
+	StdinPipe.Close()
+	err = cmd.Wait()
+	if err == nil {
+		log.Infof("Core was succesfully packed.")
+	} else {
+		err = fmt.Errorf("There was some problem packing archive. "+
+			"Error: '%v'.", err)
+	}
+	return err
+}
+
+// Unpack unpacks a tar.gz archive.
+func Unpack(tarName string) error {
+	tarPath, err := filepath.Abs(tarName)
+	if err != nil {
+		err = fmt.Errorf("There was some problem unpacking archive. "+
+			"Error: '%v'.", err)
+	}
+	if _, err := os.Stat(tarPath); err != nil {
+		err = fmt.Errorf("There was some problem unpacking archive. "+
+			"Error: '%v'.", err)
+	}
+	err = ExtractTarGz(tarName)
+	if err == nil {
+		log.Infof("Archive was succesfully unpacked. \n")
+	} else {
+		err = fmt.Errorf("There was some problem unpacking archive. "+
+			"Error: '%v'.", err)
+	}
+	return err
+}
+
+// Inspect allows user to inspect unpacked coredump.
+func Inspect(coreFolder string) error {
+	corePath, err := filepath.Abs(coreFolder)
+	if err != nil {
+		return fmt.Errorf("There was some problem inspecting archive. "+
+			"Error: '%v'.", err)
+	}
+	if _, err := os.Stat(corePath); err != nil {
+		return fmt.Errorf("There was some problem inspecting archive. "+
+			"Error: '%v'.", err)
+	}
+	content, err := util.ReadEmbedFile(corescripts, "scripts/gdb.sh")
```

Comment:
It looks like a copy-paste from `Pack`, move it to a separate function.

Reply:
Made sep. function